### PR TITLE
feat: implement error boundary for root component

### LINKED ISSUE #187: feat: implement an error boundary at root
# Feature Request

## User Story

As a developer, I want an [error boundary](http://reactjs.org/docs/error-boundaries.html) at the root component so we can catch errors and gracefully fail.

As a user, I don’t want to be confused by a blank error screen.

## Possible Solution

In order to use error boundaries in React, we will need to use a class component, since there are no hooks for catching errors yet.

## Acceptance Criteria

- [ ] Root component should catch errors and display a friendly error message.

### PR DESCRIPTION
**Pull request by knowler**
## Description

Resolves #187

This uses a class component as the root component since that’s the only was we can implement error boundaries.

Note: this modifies the ESLint rules to allow `console` error, warn, and info since those are a useful tool for communicating to whoever is debugging in a browser.

## How Has This Been Tested?

Locally; manually.

## Screenshots (jpeg or gifs if applicable):

error

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by knowler** [comment]
I’m considering refactoring this to use [bvaughn/react-error-boundary](https://github.com/bvaughn/react-error-boundary). That way we would have a reusable `ErrorBoundary` component throughout and we could convert the `RootComponent` back to be a functional component.

**Comment by JasonTheAdams** [comment]
Error boundaries are great, but that specific package doesn't look actively maintained. It hasn't been touched since March.

**Comment by knowler** [comment]
True, however, I think that since:

1. The package has a very simple purpose which requires minimal maintenance;
2. [The package is the most popular package of the type](https://github.com/search?o=desc&q=react+error+boundary&s=stars&type=Repositories);
3. [The maintainer](https://github.com/bvaughn) is a React core team member (notably the primary contributor to [React DevTools](https://github.com/facebook/react/tree/master/packages/react-devtools));
4. [The package has recognition from prominent React community members](https://github.com/bvaughn/react-error-boundary/issues/34#issue-512614594);

The package is trustworthy.

Otherwise, it would not be too difficult to write our own (perhaps with the Suspense-like API that Kent requested in point 4).

**Comment by JasonTheAdams** [comment]
Fair enough, it is simple, and even writing our own is not a big deal. It really just needs to satisfy handling one static method. Carry on. 😄

**Comment by knowler** [comment]
@JasonTheAdams I ended up going with our own `ErrorBoundary` and dropped the dependency. I put it in `@givewp/common` since it didn’t make sense to put it in `@givewp/components` since it’s not presentational and works more as a common helper component.

File: resources/js/common/components/ErrorBoundary.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+		this.state = { hasError: false, error: null };
+	}
+
+	static getDerivedStateFromError( error ) {
+		return { hasError: true, error };
+	}
+
+	render() {
+		return this.state.hasError ?
+			this.props.fallback :
```
Comment (JasonTheAdams) [inline resources/js/common/components/ErrorBoundary.jsx]: If you made `fallback` a [render prop](https://reactjs.org/docs/render-props.html) then you could pass the error message to be optionally used.

Comment (knowler) [reply resources/js/common/components/ErrorBoundary.jsx]: Good idea. I’m rebasing this, then I’ll merge with that change.

**Pull request merged**